spark/entrypoint.sh

```diff
@@ -4,4 +4,4 @@
 /opt/spark/sbin/start-worker.sh spark://spark-iceberg:7077
 /opt/spark/sbin/start-history-server.sh
 
-tail -f /dev/null
+pyspark-notebook
```

Comment:
@samredai In part 2 you have to go back to Docker/Spark blog to find the commands. This way we run the notebook right away, rendering the `docker exec -it spark-iceberg pyspark-notebook` obselete. WDYT?

Reply:
Great suggestion!

Just updated the code to do so:
```bash
fokkodriesprong@Fokkos-MacBook-Pro spark % docker run -t -i fokkospark bash
starting org.apache.spark.deploy.master.Master, logging to /opt/spark/logs/spark--org.apache.spark.deploy.master.Master-1-27c0fe8eaa61.out
starting org.apache.spark.deploy.worker.Worker, logging to /opt/spark/logs/spark--org.apache.spark.deploy.worker.Worker-1-27c0fe8eaa61.out
starting org.apache.spark.deploy.history.HistoryServer, logging to /opt/spark/logs/spark--org.apache.spark.deploy.history.HistoryServer-1-27c0fe8eaa61.out
root@27c0fe8eaa61:/opt/spark# ^C
exit
```
or:
```bash
fokkodriesprong@Fokkos-MacBook-Pro spark % docker run -t -i fokkospark notebook
starting org.apache.spark.deploy.master.Master, logging to /opt/spark/logs/spark--org.apache.spark.deploy.master.Master-1-11586dca733c.out
starting org.apache.spark.deploy.worker.Worker, logging to /opt/spark/logs/spark--org.apache.spark.deploy.worker.Worker-1-11586dca733c.out
starting org.apache.spark.deploy.history.HistoryServer, logging to /opt/spark/logs/spark--org.apache.spark.deploy.history.HistoryServer-1-11586dca733c.out
[I 18:35:53.185 NotebookApp] Writing notebook server cookie secret to /root/.local/share/jupyter/runtime/notebook_cookie_secret
[I 18:35:53.185 NotebookApp] Authentication of /metrics is OFF, since other authentication is disabled.
[W 18:35:53.349 NotebookApp] WARNING: The notebook server is listening on all IP addresses and not using encryption. This is not recommended.
[W 18:35:53.349 NotebookApp] WARNING: The notebook server is listening on all IP addresses and not using authentication. This is highly insecure and not recommended.
[I 18:35:53.353 NotebookApp] Serving notebooks from local directory: /home/iceberg/notebooks
[I 18:35:53.353 NotebookApp] Jupyter Notebook 6.4.11 is running at:
[I 18:35:53.353 NotebookApp] http://11586dca733c:8888/
[I 18:35:53.353 NotebookApp] Use Control-C to stop this server and shut down all kernels (twice to skip confirmation).
```

and also docker-compose works:
```
fokkodriesprong@Fokkos-MacBook-Pro docker-spark-iceberg % docker-compose up           
[+] Running 2/0
 ⠿ Container postgres       Created                                                                                                                                                                         0.0s
 ⠿ Container spark-iceberg  Recreated                                                                                                                                                                       0.0s
Attaching to postgres, spark-iceberg
postgres       | 
postgres       | PostgreSQL Database directory appears to contain a database; Skipping initialization
postgres       | 
postgres       | 2022-05-02 18:54:55.034 UTC [1] LOG:  starting PostgreSQL 13.4 (Debian 13.4-4.pgdg110+1) on aarch64-unknown-linux-gnu, compiled by gcc (Debian 10.2.1-6) 10.2.1 20210110, 64-bit
postgres       | 2022-05-02 18:54:55.034 UTC [1] LOG:  listening on IPv4 address "0.0.0.0", port 5432
postgres       | 2022-05-02 18:54:55.034 UTC [1] LOG:  listening on IPv6 address "::", port 5432
postgres       | 2022-05-02 18:54:55.036 UTC [1] LOG:  listening on Unix socket "/var/run/postgresql/.s.PGSQL.5432"
postgres       | 2022-05-02 18:54:55.038 UTC [26] LOG:  database system was interrupted; last known up at 2022-05-02 18:50:18 UTC
postgres       | 2022-05-02 18:54:55.084 UTC [26] LOG:  database system was not properly shut down; automatic recovery in progress
postgres       | 2022-05-02 18:54:55.085 UTC [26] LOG:  redo starts at 0/15CFE28
postgres       | 2022-05-02 18:54:55.085 UTC [26] LOG:  invalid record length at 0/15CFF10: wanted 24, got 0
postgres       | 2022-05-02 18:54:55.085 UTC [26] LOG:  redo done at 0/15CFED8
postgres       | 2022-05-02 18:54:55.093 UTC [1] LOG:  database system is ready to accept connections
spark-iceberg  | starting org.apache.spark.deploy.master.Master, logging to /opt/spark/logs/spark--org.apache.spark.deploy.master.Master-1-9a4da98f5c71.out
spark-iceberg  | starting org.apache.spark.deploy.worker.Worker, logging to /opt/spark/logs/spark--org.apache.spark.deploy.worker.Worker-1-9a4da98f5c71.out
spark-iceberg  | starting org.apache.spark.deploy.history.HistoryServer, logging to /opt/spark/logs/spark--org.apache.spark.deploy.history.HistoryServer-1-9a4da98f5c71.out
spark-iceberg  | [I 18:55:03.051 NotebookApp] Writing notebook server cookie secret to /root/.local/share/jupyter/runtime/notebook_cookie_secret
spark-iceberg  | [I 18:55:03.052 NotebookApp] Authentication of /metrics is OFF, since other authentication is disabled.
spark-iceberg  | [W 18:55:03.144 NotebookApp] WARNING: The notebook server is listening on all IP addresses and not using encryption. This is not recommended.
spark-iceberg  | [W 18:55:03.144 NotebookApp] WARNING: The notebook server is listening on all IP addresses and not using authentication. This is highly insecure and not recommended.
spark-iceberg  | [I 18:55:03.147 NotebookApp] Serving notebooks from local directory: /home/iceberg/notebooks
spark-iceberg  | [I 18:55:03.147 NotebookApp] Jupyter Notebook 6.4.11 is running at:
spark-iceberg  | [I 18:55:03.147 NotebookApp] http://9a4da98f5c71:8888/
spark-iceberg  | [I 18:55:03.147 NotebookApp] Use Control-C to stop this server and shut down all kernels (twice to skip confirmation).
spark-iceberg  | [I 18:55:07.382 NotebookApp] 302 GET / (172.19.0.1) 0.760000ms
```
I moved some of the environment variables from docker-compose to the Dockerfile itself, so the container can also be used on its own 👍🏻